scala/private/phases/phase_compile.bzl

```diff
@@ -438,7 +442,7 @@ def _try_to_compile_java_jar(
         implicit_junit_deps_needed_for_java_compilation,
     )
     providers_of_dependencies += _collect_java_providers_of(
-        _get_scalac_provider(ctx).default_classpath,
```

Comment:
note this is possibly the only interesting change in this PR since it moves something from a method call to taking it from a previous phase and still I'm confident